.github/workflows/maintainer-edit.yml

```
@@ -0,0 +1,22 @@
+name: "Check Maintainer Edit Permissions"
+on:
+  pull_request_target:
+    types:
+      - opened
+jobs:
+  PermissionsCheck:
+    env:
+      MAINTAINER_CAN_MODIFY: ${{ github.event.pull_request.maintainer_can_modify }}
+      ALLOW_LISTED: ${{ contains(fromJSON('["COLLABORATOR", "MEMBER", "OWNER"]'), github.event.pull_request.author_association) }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Comment if maintainers cannot edit
+        if: ${{ ( env.ALLOW_LISTED == false ) && ( env.MAINTAINER_CAN_MODIFY == false ) }}
```

```suggestion
        if: ${{ ( env.ALLOW_LISTED == 'false' ) && ( env.MAINTAINER_CAN_MODIFY == 'false' ) }}
```
Not 100% on this but usually env vars are strings and don't have type information. If you used the `contains()` down in the `if` you could use `== false` but since it's going to an env var, I think it will be coerced to a string.

---

Validated this works in my test repo; LGTM!